lib/3scale/backend/rack/prometheus.rb

```diff
@@ -8,7 +8,15 @@ def initialize(app)
 
         def call(env)
           began_at = Time.now.getutc
-          status, header, body = @app.call(env)
+
+          begin
+            status, header, body = @app.call(env)
+          rescue StandardError => e
+            ListenerMetrics.report_resp_code(env['REQUEST_PATH'], 500)
+            ListenerMetrics.report_response_time(env['REQUEST_PATH'], Time.now - began_at)
+            raise e
+          end
```

Comment:
This won't catch `Exception.new "abc"`, which will end in a 500 status code.

Reply:
Yeah, at first I thought that increasing the counter in situations like a signal or a syntax error is not going to be that important, but let's add it for completeness.
Changed :+1: